boot/boot.go

```diff
@@ -191,6 +201,10 @@ type bootState interface {
 	// of the update is done via bootStateUpdate's commit, that
 	// way different markSuccessful can be folded together.
 	markSuccessful(bootStateUpdate) (bootStateUpdate, error)
+
+	// getRebootBootloader returns a bootloader.RebootBootloader
+	// interface if implemented by the bootloader, nil otherwise.
+	getRebootBootloader() (bootloader.RebootBootloader, error)
```

Comment:
it's a bit unclear to me whether the best it to add this new method or incorporate returning the new info in the relevant preexisting method(s)